### PR TITLE
Roundtrip fixes

### PR DESCRIPTION
**Pull request by wesboyt**
*Issue #, if available:*
continues https://github.com/amzn/ion-js/tree/roundtripFix work after master merge.
*Description of changes:*
Removes most of the repetitive logic within roundtripEventStream.
The last stream is not strictly necessary.
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tests/unit/iontests.js [hunk 1]
```diff
@@ -273,30 +273,46 @@ define([
...
+            let textWriter = new ion.makeTextWriter();
+            textWriter.writeValues(new ion.makeReader(input));
+            textWriter.close();
+            let text = textWriter.getBytes();
+            streams.push(new es.IonEventStream(new ion.makeReader(text)));
+            let binaryWriter = new ion.makeBinaryWriter();
+            binaryWriter.writeValues(new ion.makeReader(input));
+            binaryWriter.close();
+            let binary = binaryWriter.getBytes();
+            streams.push(new es.IonEventStream(new ion.makeReader(binary)));
```
Comment (almann) [inline tests/unit/iontests.js]: Minor: Some comments in this wall of code would be nice, what is the intention here, we have a stream of the original events, followed by the events from a serialize/read as text, then etc...  Being clear about the combinatorics would be rather nice to have for the next person.

File: tests/unit/iontests.js [hunk 2]
```diff
@@ -273,30 +273,46 @@ define([
...
+
+            //second pass
+            textWriter = new ion.makeTextWriter();
+            textWriter.writeValues(new ion.makeReader(text));
+            textWriter.close();
+            streams.push(new es.IonEventStream(new ion.makeReader(textWriter.getBytes())));
```
Comment (almann) [inline tests/unit/iontests.js]: I can't help but notice that there sure is a lot of copy-paste, I would hoist the logic into a generic helper.

Perhaps something like:

```javascript
function makeStream(src, makeWriter) {
  const writer = makeWriter;
  writer.writeValues(new ion.makeReader(src));
  writer.close();
  const serialized = writer.getBytes();
  return new es.IonEventStream(new ion.makeReader(serialized));
}
```

Then the call sites would be:

```javascript
// second pass over input as serialized as text...
streams.push(makeStream(text, ion.makeTextWriter));
```

I think that would make this wall of code way easier to read.

File: tests/unit/iontests.js [hunk 3]
```diff
@@ -273,30 +273,46 @@ define([
...
 
-            let eventWriter = ion.makeTextWriter();
-            streams[0].writeEventStream(eventWriter);
-            eventWriter.close();
-            streams.push(new es.IonEventStream(new ion.makeReader(eventWriter.getBytes())));
+            binaryWriter = new ion.makeBinaryWriter();
```
Comment (almann) [inline tests/unit/iontests.js]: Should this have a `new`?  I thought `makeBinaryWriter` was a simple function not a constructor.

File: tests/unit/iontests.js (hunk 2 above)
Comment (wesboyt) [reply tests/unit/iontests.js]: done

File: tests/unit/iontests.js (hunk 1 above)
Comment (wesboyt) [reply tests/unit/iontests.js]: i hope the limited comments I added made it a bit more clear.

File: tests/unit/iontests.js (hunk 3 above)
Comment (wesboyt) [reply tests/unit/iontests.js]: doesnt seem necessary, I'll remove it

File: tests/unit/iontests.js [hunk 4]
```diff
@@ -273,46 +273,43 @@ define([
             return fs.readFileSync(path, options);
         }
 
+        function makeStream(src, makeWriter) {
+            let writer = makeWriter;
```
Comment (almann) [inline tests/unit/iontests.js]: In my feedback, I suggested that `makeWriter` be a function argument, not the `writer` itself.

If you make this actually a *function* (i.e. callable), you can streamline the code below as:

```javascript
streams.push(makeStream(input, ion.makeTextWriter));
```

And so-on.  If you choose not to take this advice, please don't have a redundant parameter called `makeWriter` that is literally just assigned to a local named `writer`.

Comment (wesboyt) [reply tests/unit/iontests.js]: I went with streams.push(makeStream(input, ion.makeTextWriter()));

**Pull request merged**